Automattic/wp-calypso
# Migration Flow Improvements: Offer 50% off for the first year on the Upgrade screen

Related to #91166 

## Proposed Changes

* Remove free trial when the introductory offer is available.
* Display the introductory offer (price, badge, and CTA).
* Use hooks instead of Redux states.

## Why are these changes being made?


* We want to offer the introductory discount to the user in the migration flow.

## Testing Instructions



* Enable the introductory offer feature in `config/development.json` (look for `migration-flow/introductory-offer` and set it to `true`) or use `.env` for this purpose: ```ACTIVE_FEATURE_FLAGS="migration-flow/introductory-offer"```
* Go to https://wordpress.com/me/account and switch the language to English.
* Run Calypso.
* Open `calypso.localhost:3000/start` and take the steps to reach the upgrade step (choose a domain, choose a plan, check import content in goals, choose import everything and do it on your own).
* On the upgrade step, make sure:
  * you see the discounted price and the original price next to it;
  * you see the text explaining the payment steps and taxes ("per month, $150 billed annually for the first year, $300 afterwards, excl. taxes" for USD).
  * you see the "One time offer" badge above the prices;
  * you see the CTA "Get the plan and migrate";
  * you don't see the free trial button;
  * you don't see the period switcher (annual/monthly);
  * the page looks as expected on mobile.
* Click on "Get the plan and migrate" and make sure the introductory offer is applied on the Checkout.
* Disable the feature flag (`migration-flow/introductory-offer`).
* Repeat the the steps until you get to the Upgrade step
* On the upgrade step, make sure:
  * you see the standard price;
  * CTA is "Upgrade and migrate";
  * there is the period switcher and it works as expected;
  * there is the free trial button;
  * the page looks as expected on mobile.
* Click on "Get the plan and migrate" and make sure the introductory offer is not applied on the Checkout.
* Go to https://wordpress.com/me/account and switch the language to a non-English one.
* Repeat the the steps until you get to the Upgrade step
* On the upgrade step, make sure you don't see untranslated strings.
* Change the language back to English.
* Turn off the introductory offer feature in `config/development.json`.
* Repeat the steps until you get to the upgrade step.
* On the upgrade step, make sure:
  * you see the standard price;
  * CTA is "Upgrade and migrate";
  * there is the period switcher and it works as expected;
  * there is the free trial button;
  * the page looks as expected on mobile.

## Screenshots
<img width="858" alt="Migration Flow Improvements" src="https://github.com/Automattic/wp-calypso/assets/329356/9b02ab4a-cceb-41b0-a4e8-409bb1d00c84">

## Pre-merge Checklist
